### ISSUE TITLE
(Proposal) (BIG) Built-in level detail view

### ISSUE THREAD

**Issue by JOELwindows7** [Bithack/principia#37]
# Have principia-web level explorer right in the app.
From the website, level details description complete with `play` `edit` `like` button. Not with in-built browser, all GUI in it.

Basically the entire website a GUI of the app itself, so the experience becomes seamless, no need browser anymore.

and when inside `play`ed level, there's an option to pop up the level details again with `play` `edit` `like` button at anytime when press back, or through top right menu.

**Comment by rollerozxa** [comment]
So basically, the entire community site implemented inside of the game? Sounds like an interesting idea but a very hard one to pull off in its entirety. Apparatus had something like this planned (HoneycombCommunityActivity) but that made use of Android's UI layouting system and was never finished.

**Comment by JOELwindows7** [comment]
> So basically, the entire community site implemented inside of the game? Sounds like an interesting idea but a very hard one to pull off in its entirety. Apparatus had something like this planned (HoneycombCommunityActivity) but that made use of Android's UI layouting system and was never finished.

Yeah basically it. I just, like convenience & seamlessness and yeah, you see that's a big thing to do so.

It's here, just to remind to eventually make so, slowly, idk..

**Comment by griffi-gh** [comment]
Maybe I'll add this later if the imgui backend gets merged. (currently wip, most dialogs are missing)

**Comment by rollerozxa** [comment]
Began work on this a while back. Fetches a list of levels from a new (not yet public) API on the community site, and then displays them in a level browser dialog.

<img width="1046" height="785" alt="Image" src="https://github.com/user-attachments/assets/6d3eb910-3d07-4618-a1d7-1e617d56b641" />

What's left of implementing is basically: Fetching level thumbnails, adding the filter options that are accessible on the browse page, adding a search field, and adding a second level details dialog that will show when you click on a level.

The idea of it is not to replace the community site but to offer a more tightly integrated alternative that gives you enough to just browse and play levels without browser integration, which would be useful on platforms such as the Switch.
